nav2_costmap_2d/plugins/obstacle_layer.cpp

```diff
@@ -549,19 +551,23 @@ ObstacleLayer::updateBounds(
         continue;
       }
 
-      // compute the squared distance from the hitpoint to the pointcloud's origin
-      double sq_dist =
-        (px -
-        obs.origin_.x) * (px - obs.origin_.x) + (py - obs.origin_.y) * (py - obs.origin_.y) +
-        (pz - obs.origin_.z) * (pz - obs.origin_.z);
+      // compute the distance from the hitpoint to the pointcloud's origin
+      // Calculate the distance in cell space to match the ray trace algorithm
+      // used for clearing obstacles (see Costmap2D::raytraceLine).
+      unsigned int x0, y0, x1, y1;
+      worldToMap(obs.origin_.x, obs.origin_.y, x0, y0);
+      worldToMap(px, py, x1, y1);
+      const uint64_t sq_dist =
+        static_cast<int64_t>(x1 - x0) * static_cast<int64_t>(x1 - x0) +
```

Comment:
I think you need to cast the unsigned ints to be signed before subtraction. It can wrap around on most systems if x1 < x0